src/audio/component.c

```diff
@@ -159,17 +159,20 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 		}
 		break;
 	case COMP_TRIGGER_STOP:
-	case COMP_TRIGGER_XRUN:
 		if (dev->state == COMP_STATE_ACTIVE ||
 		    dev->state == COMP_STATE_PAUSED) {
 			dev->state = COMP_STATE_PREPARE;
 		} else {
 			trace_comp_error("comp_set_state() error: "
 					 "wrong state = %u, "
-					 "COMP_TRIGGER_STOP/XRUN", dev->state);
+					 "COMP_TRIGGER_STOP", dev->state);
 			ret = -EINVAL;
 		}
 		break;
+	case COMP_TRIGGER_XRUN:
```

Comment:
@keyonjie does the state transition diagram need updated in the component header ?

Reply:
correct, let me update it.